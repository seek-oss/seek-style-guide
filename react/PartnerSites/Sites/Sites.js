import styles from './Sites.less';

import React from 'react';
import PropTypes from 'prop-types';

const linksObject = {
  AU: {
    jobsUrl: 'https://www.seek.com.au',
    learningUrl:
      'https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link',
    businessUrl: 'https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus',
    volunteerUrl: 'https://www.volunteer.com.au/?tracking=SKMAU:main+header'
  },
  NZ: {
    jobsUrl: 'https://www.seek.co.nz',
    learningUrl:
      'https://www.seeklearning.co.nz/?campaigncode=seek_banner_29&sc_trk=skj-courses-link',
    businessUrl:
      'https://www.seekbusiness.com.au/?site=nz&tracking=sk:main:nz:nav:bus',
    volunteerUrl: 'https://seekvolunteer.co.nz/?tracking=SKMNZ:main+header'
  }
};

const err = () => {
  const e = new Error('Link key not found');
  e.name = 'PartnerSiteError';
  throw e;
};

const localeLens = links => locale => (links ? links[locale] : err());
const linkLens = locale => link =>
  (typeof locale[link] === 'string' && locale[link]) || err();

export default function Sites({ locale, linkRenderer, activeSite }) {
  const linkView = linkLens(localeLens(linksObject)(locale));

  const isJobsActive = activeSite === 'Jobs';
  const isCoursesActive = activeSite === 'Courses';
  const isBusinessesActive = activeSite === 'Businesses for sale';
  const isVolunteeringActive = activeSite === 'Volunteering';

  return (
    <nav aria-label="SEEK sites" role="navigation">
      <ul className={styles.list}>
        <li>
          {isJobsActive ? (
            <span
              className={`${styles.link} ${styles.link_isActive}`}
              title="SEEK Jobs"
            >
              Jobs
            </span>
          ) : (
            linkRenderer({
              'data-analytics': 'header:jobs',
              className: `${styles.link} ${styles.link_isJobs}`,
              href: linkView('jobsUrl'),
              title: 'SEEK Jobs',
              children: 'Jobs'
            })
          )}
        </li>
        <li>
          {isCoursesActive ? (
            <span
              className={`${styles.link} ${styles.link_isActive}`}
              title="SEEK Learning"
            >
              Courses
            </span>
          ) : (
            linkRenderer({
              'data-analytics': 'header:courses',
              className: `${styles.link} ${styles.link_isLearning}`,
              href: linkView('learningUrl'),
              title: 'SEEK Learning',
              children: 'Courses'
            })
          )}
        </li>
        <li>
          {isBusinessesActive ? (
            <span
              className={`${styles.link} ${styles.link_isActive}`}
              title="SEEK Business"
            >
              Businesses for sale
            </span>
          ) : (
            linkRenderer({
              'data-analytics': 'header:business+for+sale',
              className: `${styles.link} ${styles.link_isBusiness}`,
              href: linkView('businessUrl'),
              title: 'SEEK Business',
              children: 'Businesses for sale'
            })
          )}
        </li>
        <li>
          {isVolunteeringActive ? (
            <span
              className={`${styles.link} ${styles.link_isActive}`}
              title="SEEK Volunteer"
            >
              Volunteering
            </span>
          ) : (
            linkRenderer({
              'data-analytics': 'header:volunteering',
              className: `${styles.link} ${styles.link_isVolunteering}`,
              href: linkView('volunteerUrl'),
              title: 'SEEK Volunteer',
              children: 'Volunteering'
            })
          )}
        </li>
      </ul>
    </nav>
  );
}

Sites.propTypes = {
  locale: PropTypes.oneOf(['AU', 'NZ']).isRequired,
  linkRenderer: PropTypes.func.isRequired,
  activeSite: PropTypes.oneOf([
    'Jobs',
    'Courses',
    'Businesses for sale',
    'Volunteering'
  ])
};

Sites.defaultProps = {
  activeSite: null
};
