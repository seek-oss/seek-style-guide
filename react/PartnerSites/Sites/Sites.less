@import (reference) "~seek-style-guide/theme";

.list {
  display: flex;
}

.link {
  .nav_link();
  .rawText(@small-type-scale);
  height: (@row-height * 5);
  margin: 0 @gutter-width;
  display: flex;
  align-items: center;
  color: @sk-mid-gray-dark;
}

.link_isActive {
  box-shadow: inset 0 -2px @sk-black;
  font-weight: @sk-bold;
  color: @sk-black;

  &:hover {
    color: @sk-black;
    cursor: auto;
  }
}

.link_isJobs:hover {
  color: @sk-link-on-mid-gray-light;
}

.link_isLearning:hover {
  color: @sk-learning-dark-on-gray-light;
}

.link_isBusiness:hover {
  color: @sk-business-on-gray-light;
}

.link_isVolunteering:hover {
  color: @sk-pink-on-gray-light;
}
