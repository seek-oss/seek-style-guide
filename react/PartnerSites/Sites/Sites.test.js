import React from 'react';
import { render } from 'enzyme';

import Sites from './Sites';

const renderSites = props => render(<Sites {...props} />);
const linkRenderer = props => <a {...props} />;

describe('Sites:', () => {
  describe('locale states:', () => {
    it('should render with locale of AU', () => {
      expect(renderSites({ locale: 'AU', linkRenderer })).toMatchSnapshot();
    });

    it('should render with locale of NZ', () => {
      expect(renderSites({ locale: 'NZ', linkRenderer })).toMatchSnapshot();
    });
  });
  describe('active product states:', () => {
    it('should render with no link as active by default', () => {
      expect(renderSites({ locale: 'AU', linkRenderer })).toMatchSnapshot();
    });
    it('should render with jobs link as active', () => {
      expect(
        renderSites({ locale: 'AU', linkRenderer, activeProduct: 'Jobs' })
      ).toMatchSnapshot();
    });
    it('should render with courses link as active', () => {
      expect(
        renderSites({ locale: 'AU', linkRenderer, activeProduct: 'Courses' })
      ).toMatchSnapshot();
    });
    it('should render with businesses link as active', () => {
      expect(
        renderSites({
          locale: 'AU',
          linkRenderer,
          activeProduct: 'Businesses for sale'
        })
      ).toMatchSnapshot();
    });
    it('should render with volunteering link as active', () => {
      expect(
        renderSites({
          locale: 'AU',
          linkRenderer,
          activeProduct: 'Volunteering'
        })
      ).toMatchSnapshot();
    });
  });
});
