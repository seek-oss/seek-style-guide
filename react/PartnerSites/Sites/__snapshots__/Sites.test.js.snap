// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sites: active product states: should render with businesses link as active 1`] = `
<nav
  aria-labelledby="Sites"
  role="navigation"
>
  <span
    class="ScreenReaderOnly__root"
  >
    <h1
      id="Sites"
    >
      SEEK Sites
    </h1>
  </span>
  <ul
    class="Sites__list"
  >
    <li>
      <a
        class="Sites__link Sites__link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Sites: active product states: should render with courses link as active 1`] = `
<nav
  aria-labelledby="Sites"
  role="navigation"
>
  <span
    class="ScreenReaderOnly__root"
  >
    <h1
      id="Sites"
    >
      SEEK Sites
    </h1>
  </span>
  <ul
    class="Sites__list"
  >
    <li>
      <a
        class="Sites__link Sites__link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Sites: active product states: should render with jobs link as active 1`] = `
<nav
  aria-labelledby="Sites"
  role="navigation"
>
  <span
    class="ScreenReaderOnly__root"
  >
    <h1
      id="Sites"
    >
      SEEK Sites
    </h1>
  </span>
  <ul
    class="Sites__list"
  >
    <li>
      <a
        class="Sites__link Sites__link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Sites: active product states: should render with no link as active by default 1`] = `
<nav
  aria-labelledby="Sites"
  role="navigation"
>
  <span
    class="ScreenReaderOnly__root"
  >
    <h1
      id="Sites"
    >
      SEEK Sites
    </h1>
  </span>
  <ul
    class="Sites__list"
  >
    <li>
      <a
        class="Sites__link Sites__link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Sites: active product states: should render with volunteering link as active 1`] = `
<nav
  aria-labelledby="Sites"
  role="navigation"
>
  <span
    class="ScreenReaderOnly__root"
  >
    <h1
      id="Sites"
    >
      SEEK Sites
    </h1>
  </span>
  <ul
    class="Sites__list"
  >
    <li>
      <a
        class="Sites__link Sites__link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Sites: locale states: should render with locale of AU 1`] = `
<nav
  aria-labelledby="Sites"
  role="navigation"
>
  <span
    class="ScreenReaderOnly__root"
  >
    <h1
      id="Sites"
    >
      SEEK Sites
    </h1>
  </span>
  <ul
    class="Sites__list"
  >
    <li>
      <a
        class="Sites__link Sites__link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.com.au"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isLearning"
        data-analytics="header:courses"
        href="https://www.seek.com.au/learning/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?tracking=sk:main:au:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://www.volunteer.com.au/?tracking=SKMAU:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Sites: locale states: should render with locale of NZ 1`] = `
<nav
  aria-labelledby="Sites"
  role="navigation"
>
  <span
    class="ScreenReaderOnly__root"
  >
    <h1
      id="Sites"
    >
      SEEK Sites
    </h1>
  </span>
  <ul
    class="Sites__list"
  >
    <li>
      <a
        class="Sites__link Sites__link_isJobs"
        data-analytics="header:jobs"
        href="https://www.seek.co.nz"
        title="SEEK Jobs"
      >
        Jobs
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isLearning"
        data-analytics="header:courses"
        href="https://www.seeklearning.co.nz/?campaigncode=seek_banner_29&sc_trk=skj-courses-link"
        title="SEEK Learning"
      >
        Courses
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isBusiness"
        data-analytics="header:business+for+sale"
        href="https://www.seekbusiness.com.au/?site=nz&tracking=sk:main:nz:nav:bus"
        title="SEEK Business"
      >
        Businesses for sale
      </a>
    </li>
    <li>
      <a
        class="Sites__link Sites__link_isVolunteering"
        data-analytics="header:volunteering"
        href="https://seekvolunteer.co.nz/?tracking=SKMNZ:main+header"
        title="SEEK Volunteer"
      >
        Volunteering
      </a>
    </li>
  </ul>
</nav>
`;
